File: app/src/main/java/com/example/navigationdrawerapp/CollapseViewHolder.java
package com.example.navigationdrawerapp;

import android.view.View;

import com.example.navigationdrawerapp.databinding.CollapseBinder;

public class CollapseViewHolder extends BaseViewHolder {

    private final String TAG = CollapseViewHolder.class.getSimpleName();

    private final CollapseBinder binder;

    public CollapseViewHolder(CollapseBinder binder, DrawerListener listener) {
        super(binder.getRoot(), listener);
        this.binder = binder;
        binder.getRoot().setId(-1);
        binder.getRoot().setVisibility(View.GONE);
    }

    @Override
    void bindDataToViewHolder(DrawerModel model, int position) {
        //region Input Data
        binder.setHolder(model);
        binder.setPosition(position);
        binder.executePendingBindings();
        //binder.textView.setText(model.getText());
        //endregion
    }
}

File: app/src/main/java/com/example/navigationdrawerapp/CollapseViewHolder.kt
package com.example.navigationdrawerapp

import android.view.View
import com.example.navigationdrawerapp.databinding.CollapseBinder

class CollapseViewHolder : BaseViewHolder {

    companion object {
        private val TAG = CollapseViewHolder::class.java.getSimpleName()
    }

    private val binder : CollapseBinder

    constructor(binder : CollapseBinder, listeners : DrawerListener) : super(binder.getRoot(), listeners) {
        this.binder = binder
        binder.getRoot().setId(-1)
        binder.getRoot().setVisibility(View.GONE)
    }

    public override fun bindDataToViewHolder(model : DrawerModel, position : Int) {
        //region Input Data
        binder.setHolder(model)
        binder.setPosition(position)
        binder.executePendingBindings()
        binder.textView.setText(model.text)
        //endregion
    }
}